### PR TITLE
add makefile and improve github actions

### PR DESCRIPTION
**Pull request by mmulholla**
Add makefile with standard build tasks

Add go fmt and go mod tidy to build

Extend github actions to be used on Pull requests.

### CODE REVIEW

File: Makefile
Comment (baijum) [inline Makefile:22]: Can we add `-v` flag for verbose output?

Comment (mmulholla) [reply Makefile]: Added.

**Comment by mmulholla** [comment]
Note the build will not pass until this PR is merged: https://github.com/redhat-certification/chart-verifier/pull/51

File: Makefile
Comment (baijum) [inline Makefile:18]: Do we need both these commands? Now there are two binaries, right?

**Comment by mmulholla** [comment]
@baijum all working now - ready for final review

File: Makefile
Comment (mmulholla) [reply Makefile]: True, removed the first.

File: .gitignore
Comment (baijum) [inline .gitignore:3]: This may not be required as the binary output goes inside the `out` directory.

**Comment by baijum** [comment]
@mmulholla The title still has "DO NOT MERGE". Please change it it's changed.

File: .gitignore
Comment (mmulholla) [reply .gitignore]: removed

**Review by baijum — APPROVED** [review]
LGTM

**Pull request merged**
